packages/components/src/Tabs/Tabs.test.tsx

```diff
@@ -17,32 +17,86 @@ const omelet = (
   </Tabs>
 );
 
-it("renders Tabs", () => {
-  const { container } = render(omelet);
-  expect(container).toMatchSnapshot();
-});
+describe("Tabs", () => {
+  it("renders Tabs", () => {
+    const { container } = render(omelet);
+    expect(container).toMatchSnapshot();
+  });
 
-test("it should switch tabs", () => {
-  const { getByText, queryByText } = render(omelet);
+  it("should switch tabs", () => {
+    const { getByText, queryByText } = render(omelet);
 
-  expect(queryByText("🍳")).toBeTruthy();
-  expect(queryByText("🧀")).toBeFalsy();
+    expect(queryByText("🍳")).toBeTruthy();
+    expect(queryByText("🧀")).toBeFalsy();
 
-  fireEvent.click(getByText("Cheese"));
-  expect(queryByText("🍳")).toBeFalsy();
-  expect(queryByText("🧀")).toBeTruthy();
+    fireEvent.click(getByText("Cheese"));
+    expect(queryByText("🍳")).toBeFalsy();
+    expect(queryByText("🧀")).toBeTruthy();
 
-  fireEvent.click(getByText("Eggs"));
-  expect(queryByText("🍳")).toBeTruthy();
-  expect(queryByText("🧀")).toBeFalsy();
-});
+    fireEvent.click(getByText("Eggs"));
+    expect(queryByText("🍳")).toBeTruthy();
+    expect(queryByText("🧀")).toBeFalsy();
+  });
+
+  it("should handle tab onClick", () => {
+    const { getByText } = render(omelet);
+    count = 0;
+
+    fireEvent.click(getByText("Cheese"));
+    expect(count).toBe(1);
+    fireEvent.click(getByText("Cheese"));
+    expect(count).toBe(2);
+  });
+
+  it("calls the onTabChange callback after a tab is clicked", () => {
+    const onTabChange = jest.fn();
+    const { getByText } = render(
+      <Tabs onTabChange={onTabChange}>
+        <Tab label="Eggs">
+          <p>🍳</p>
+          <p>Eggs</p>
+        </Tab>
+        <Tab label="Cheese">
+          <p>🧀</p>
+        </Tab>
+      </Tabs>,
+    );
+
+    fireEvent.click(getByText("Cheese"));
+    expect(onTabChange).toHaveBeenCalledWith(1, 0);
+  });
+
+  it("sets the active tab on mount", () => {
+    const { queryByText } = render(
+      <Tabs defaultTab={1}>
+        <Tab label="Eggs">
+          <p>🍳</p>
+          <p>Eggs</p>
+        </Tab>
+        <Tab label="Cheese">
+          <p>🧀</p>
+        </Tab>
+      </Tabs>,
+    );
+
+    expect(queryByText("🍳")).toBeFalsy();
+    expect(queryByText("🧀")).toBeTruthy();
```

Comment:
```suggestion
    expect(queryByText("🍳")).not.toBeInTheDocument();
    expect(queryByText("🧀")).toBeInTheDocument();
```

the `.not.toBeInTheDocument` might be a bit weird to think about in terms of queryBy, but it's pretty much the same test in a more descriptive manner

Reply:
This is good to know. I'm working on improving my jest-fu and this helps. Thanks!

Reply:
Nice! unfortunately. can't use that in JO just yet. They're planning to add it so stay tuned.

Also, rotating to Atlantis is a possibility if you want more jest-fu and front-end tricksz 😉